libs/ts-rest/core/src/lib/client.ts

```diff
@@ -202,45 +203,30 @@ export const getRouteQuery = <TAppRoute extends AppRoute>(
   };
 };
 
-const createNewProxy = (router: AppRouter, args: ClientArgs) => {
-  return new Proxy(
-    {},
-    {
-      get: (target, propKey): any => {
-        if (typeof propKey === 'string' && propKey in router) {
-          const subRouter = router[propKey];
-
-          if (isAppRoute(subRouter)) {
-            // If the current router.X is a route, return a function to handle the users args
-            return getRouteQuery(subRouter, args);
-          } else {
-            return createNewProxy(subRouter, args);
-          }
-        }
-
-        return createNewProxy(router, args);
-      },
-    }
-  );
-};
-
 export type InitClientReturn<T extends AppRouter> = RecursiveProxyObj<T>;
 
 export const initClient = <T extends AppRouter>(
   router: T,
   args: ClientArgs
 ): InitClientReturn<T> => {
-  const proxy = createNewProxy(router, args);
-
-  return proxy as InitClientReturn<T>;
+  return Object.fromEntries(
+    Object.entries(router).map(([key, subRouter]) => {
+      if (isAppRoute(subRouter)) {
+        return [key, getRouteQuery(subRouter, args)];
+      } else {
+        return [key, initClient(subRouter, args)];
```

Comment:
Really minor thing, but you don't need to wrap an else around this return.

Reply:
True, it's technically redundant, but it depends on the context and how it improves readability. Dropping the else makes sense when you want to return early or are using a guard pattern. Semantically, in this case, the if/else helps readability. It's a style preference at the end of the day.

Reply:
Personally, I like exit early patterns, but this looks fine in this case.